resources/js/features/messages/components/CreateMessageThreadForm/useCreateMessageThreadForm.ts

```diff
@@ -0,0 +1,66 @@
+import { zodResolver } from '@hookform/resolvers/zod';
+import { router } from '@inertiajs/react';
+import { useMutation } from '@tanstack/react-query';
+import axios, { type AxiosError } from 'axios';
+import { useForm } from 'react-hook-form';
+import { useTranslation } from 'react-i18next';
+import { z } from 'zod';
+
+import { toastMessage } from '@/common/components/+vendor/BaseToaster';
+import { preProcessShortcodesInBody } from '@/common/utils/shortcodes/preProcessShortcodesInBody';
+
+const formSchema = z.object({
+  recipient: z.string().min(3).max(20),
+  title: z.string().min(2).max(255),
+  body: z.string().min(1).max(60_000),
+});
+type FormValues = z.infer<typeof formSchema>;
+
+export function useCreateMessageThreadForm(defaultValues: Partial<FormValues>) {
+  const { t } = useTranslation();
+
+  const form = useForm<FormValues>({
+    resolver: zodResolver(formSchema),
+    defaultValues: {
+      recipient: '',
+      title: '',
+      body: '',
+      ...defaultValues,
+    },
+  });
+
+  const mutation = useMutation({
+    mutationFn: (formValues: FormValues) => {
+      const normalizedPayload = {
+        ...formValues,
+        body: preProcessShortcodesInBody(formValues.body),
+      };
+
+      return axios.post<{ threadId: number }>(route('api.message.store'), normalizedPayload);
+    },
+  });
+
+  const onSubmit = async (formValues: FormValues) => {
+    await toastMessage.promise(mutation.mutateAsync(formValues), {
+      loading: t('Submitting...'),
+      success: ({ data }) => {
+        router.visit(route('message-thread.show', { messageThread: data.threadId }));
```

Comment:
This seems to be redirecting to a popup window?
![image](https://github.com/user-attachments/assets/93aa0190-144a-4005-81c2-3ffa79cb3074)

Reply:
Ah, I think this may be automatically fixed by https://github.com/RetroAchievements/RAWeb/pull/3206. This is another case (similar to what happened with the Report Issue page) where an Inertia route is redirecting to a Blade route.

Reply:
Confirmed this should be fixed in latest as a result of #3206 being merged.